tasks/sql-connector/main.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Union
+
+import logging
+import csv
+import zipfile
+
+from mysql.connector import CMySQLConnection
+from psycopg2.extensions import connection
+
+import mysql.connector
+import psycopg2
+
+from coretex import currentTaskRun, CustomDataset
+
+
+def connectMysqlDatabase(connectionConfig: dict[str, str]) -> CMySQLConnection:
+    logging.info(f">> [SQL Connector] Connecting with MySQL database \"{connectionConfig["database"]}\"...")
+
+    try:
+        conn = mysql.connector.connect(**connectionConfig)
+    except mysql.connector.errors.Error as e:
+        logging.error(f">> [SQL Connector] Error while connecting to database: {e}")
+
+    return conn
+
+
+def connectPostgresqlDatabase(connectionConfig: dict[str, str]) -> connection:
+    logging.info(f">> [SQL Connector] Connecting with PostgreSQL database \"{connectionConfig["database"]}\"...")
+
+    try:
+        conn = psycopg2.connect(**connectionConfig)
+    except psycopg2._psycopg.Error as e:
+        logging.error(f">> [SQL Connector] Error while connecting to database: {e}")
+
+    return conn
+
+
+def fetchAllData(conn: Any, dataset: CustomDataset, queryGetTables: str, queryGetRows: str) -> None:
+    cursor = conn.cursor()
+    cursor.execute(queryGetTables)
+    tables = cursor.fetchall()
+    tables = [table[0] for table in tables]
+
+    for table in tables:
+        tableData = []
```

Comment:
When defining empty lists and dictionaires typing should be added.
For example: `listVariable: list[str] = []`